qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +163,139 @@ def _add_variable(self,
+        variable = Variable(self, name, lowerbound, upperbound, vartype)
+        self.variables.append(variable)
+        return variable
+
+    def _var_dict(self,
+                  lowerbound: Union[float, int] = 0,
+                  upperbound: Union[float, int] = INFINITY,
+                  vartype: VarType = VarType.CONTINUOUS,
+                  name: Optional[str] = None,
+                  formatter: Optional[str] = None,
+                  keys: Union[int, Sequence] = 1) -> Dict[str, Variable]:
+        """
+        Adds a positive number of variables to the variable list and index and returns a
+        dictionary mapping the variable names to their instances. If 'formatter' is present,
+        the next 'var_count' available indices are substituted into 'formatter' and appended
+        to 'name'.
 
         Args:
-            lowerbound: The lowerbound of the variable.
-            upperbound: The upperbound of the variable.
-            vartype: The type of the variable.
-            name: The name of the variable.
+            lowerbound: The lower bound of the variable(s).
+            upperbound: The upper bound of the variable(s).
+            vartype: The type of the variable(s).
+            name: The name(s) of the variable(s).
+            formatter: The format used to name/index the variable(s).
+            keys: If keys: int, it is interpreted as the number of variables to construct.
+                  Otherwise, the elements of the sequence are converted to strings via 'str' and
+                  passed to the formatter.
 
         Returns:
-            The added variable.
+            A dictionary mapping the variable names to variable instances.
 
         Raises:
             QiskitOptimizationError: if the variable name is already taken.
+            QiskitOptimizationError: if less than one variable instantiation is attempted.
+            QiskitOptimizationError: if the formatter has more than one substitution or a
+                                     nested substitution.
+        """
+        return dict(
+            zip(*self._add_variables(lowerbound, upperbound, vartype, name, formatter, keys)))
+
+    def _var_list(self,
+                  lowerbound: Union[float, int] = 0,
+                  upperbound: Union[float, int] = INFINITY,
+                  vartype: VarType = VarType.CONTINUOUS,
+                  name: Optional[str] = None,
+                  formatter: Optional[str] = None,
+                  keys: Union[int, Sequence] = 1) -> List[Variable]:
+        """
+        Adds a positive number of variables to the variable list and index and returns a
+        list of variable instances.
 
+        Args:
+            lowerbound: The lower bound of the variable(s).
+            upperbound: The upper bound of the variable(s).
+            vartype: The type of the variable(s).
+            name: The name(s) of the variable(s).
+            formatter: The format used to name/index the variable(s).
```

Comment:
As for the parameter name `formatter`, I think `key_format` or `var_format` or just `format` sounds good. Formatter sounds like a function to convert a format into a string. You also write the comment as "The format used to name/index the variable(s)."